### PR TITLE
JDBC E-COMERCE

### PR DESCRIPTION
**Pull request by phanirama**

### CODE REVIEW

File: src/main/java/jdbcecomarce/Connections.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.mysql.cj.Query;
+import com.mysql.cj.protocol.Resultset;
+import com.mysql.cj.protocol.a.ResultsetFactory;
+
+public class Connections {
+
+	final static String DB_URL = "jdbc:mysql://localhost:3306/employee";
+	final static String USER = "root";
+	final static String PASSWord = "Phanirama@1997";
```
Comment (admin-shapestone) [inline src/main/java/jdbcecomarce/Connections.java]: Make constant name as per the sandard

File: src/main/java/jdbcecomarce/Connections.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	}
+
+	public static void createTables() throws SQLException {
+		Connection connection = getConnetion1();
+
+		Statement statement = connection.createStatement();
+
+		String query = "DROP TABLE IF EXISTS passengers";
+		String query1 = "DROP TABLE IF EXISTS bookingss";
+		statement.execute(query);
```
Comment (admin-shapestone) [inline src/main/java/jdbcecomarce/Connections.java]: is this dropping tables properly ???

File: src/main/java/jdbcecomarce/Connections.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			statement1.executeUpdate();
+		}
+
+		System.out.println(" Data Inserted successfully");
+	}
+
+	public static void readTables(int customerId) throws SQLException {
+
+		Connection connection = getConnetion1();
+		String sql = "select * from customers where customerId=?";
```
Comment (admin-shapestone) [inline src/main/java/jdbcecomarce/Connections.java]: add purchanses information, meaning join tables and get the data

File: src/main/java/jdbcecomarce/LaptopPurchases.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package jdbcecomarce;
+
+public class LaptopPurchases {
+
+	private int customerId;
+	private int purchaseId;
+	private int quantity;
+	private String itemPurchased;
+	private double price;
+	private String dateOfPurchase;
```
Comment (admin-shapestone) [inline src/main/java/jdbcecomarce/LaptopPurchases.java]: change this type from string to Date

**Review by admin-shapestone — CHANGES_REQUESTED** [review]

File: src/main/java/jdbcecomarce/LaptopPurchases.java (hunk 4 above)
Comment (phanirama) [reply src/main/java/jdbcecomarce/LaptopPurchases.java]: ok sir

File: src/main/java/jdbcecomarce/Connections.java (hunk 2 above)
Comment (phanirama) [reply src/main/java/jdbcecomarce/Connections.java:37]: yes sir

Comment (phanirama) [reply src/main/java/jdbcecomarce/Connections.java]: i changed it into proper query now it's working ,sir

File: src/main/java/jdbcecomarce/Connections.java (hunk 3 above)
Comment (phanirama) [reply src/main/java/jdbcecomarce/Connections.java]: done sir

File: src/main/java/jdbcecomarce/Connections.java [hunk 5]
```diff
@@ -32,32 +32,32 @@ public static void createTables() throws SQLException {
 
 		Statement statement = connection.createStatement();
 
-		String query = "DROP TABLE IF EXISTS passengers";
-		String query1 = "DROP TABLE IF EXISTS bookingss";
+		String query = "DROP TABLE IF EXISTS customers";
+		String query1 = "DROP TABLE IF EXISTS purchases";
 		statement.execute(query);
```
Comment (admin-shapestone) [inline src/main/java/jdbcecomarce/Connections.java]: I don't think it will work

**Review by admin-shapestone — CHANGES_REQUESTED** [review]

**Review by admin-shapestone — APPROVED** [review]

**Pull request merged**